edx_rbac/fields.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+"""
+Fields to be used for djangoapps extending edx_rbac.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django import forms
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+
+
+class UserFromEmailField(forms.EmailField):
+    """
+    Custom Form Field class for selecting users by entering an email.
+
+    Meant to be used for models with foreign keys to the user table.
+    """
+
+    def clean(self, value):
+        """
+        Override for UserFromEmailField clean method.
+        """
+        try:
+            user = User.objects.get(email=value)
```

Comment:
i'm realizing right now that we might need to change this to use settings.AUTH_USER_MODEL instead of User so that it is compatible with ecommerce and the analytics api. we might consider updating it at the model level as well.

Reply:
Ah yes, that is correct. Irfan mentioned this to me late last week. Do we want to do it on this PR or on your main one?

Reply:
Made this change here 👍